### PR TITLE
fix(dialpad): empty address fix

### PR DESCRIPTION
**Pull request by ravicra2**
Dial pad had two issues (screenshots attached in the below Jira):
1. Empty address for a few directory contacts.
2. Empty object created without an address for a few contacts. 
Screenshot 2024-04-30 at 11 42 20 AM

Both issues are addressed as part of this PR.
Screenshot 2024-04-30 at 11 42 38 AM

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-509710

### CODE REVIEW

**Review by akulakum — COMMENTED** [review]
Have we taken care of the channel members feature and channel extension feature, using the same components that we used for the dialpad feature?

**Review by akulakum — COMMENTED** [review]
Can we also add a filter check for the same email and phoneNumber address in the speed dial feature? This is because we are using the contacts and users APIs for both the speed dial and dial pad features.

File: packages/node_modules/@webex/widget-number-pad/src/SearchContacts/SearchContactsItem.tsx [hunk 1]
```diff
@@ -51,7 +51,10 @@ export const SearchContactsItem = ({
 
   const [imageUrl] = useLazyLoadAvatarUrl(user.id, user.fetchAvatarUrl);
 
-  const callables = [...user.phoneNumbers, ...user.emailAddresses];
+  const [callDisabled, setCallDisabled] = useState<boolean>(false);
+  const filteredEmails = user.emailAddresses.length > 1 ? user.emailAddresses.filter(email => email.address !== undefined && email.address !== null) : user.emailAddresses;
+  const callables = [...user.phoneNumbers, ...filteredEmails];
```
[the remark this answers is not in the document]
Comment (ravicra2) [reply packages/node_modules/@webex/widget-number-pad/src/SearchContacts/SearchContactsItem.tsx:56]: phoneNumbers keys are not duplicated, but emailAddresses key are duplicated and second duplicated key has no value.

We are using spread operator to add all the objects including phoneNumbers and emailAddresses so always the values will be either null or undefined, it doesn’t create empty string value.

Has i have mentioned earlier there are two fixes in this PR, for second duplicated key with no value is the check user.emailAddresses.length > 1

**Comment by ravicra2** [comment]
> Can we also add a filter check for the same email and phoneNumber address in the speed dial feature? This is because we are using the contacts and users APIs for both the speed dial and dial pad features.



> Have we taken care of the channel members feature and channel extension feature, using the same components that we used for the dialpad feature?

If channel members and channel extension is using same components this will work.

**Comment by ravicra2** [comment]
> Can we also add a filter check for the same email and phoneNumber address in the speed dial feature? This is because we are using the contacts and users APIs for both the speed dial and dial pad features.

Speed Dial search contacts has different behaviour we don’t make an audio or video call there, after searching a contact we try to add the contact where in the phone field value will be undefined, need input for this how to tackle from Ivy and Akshay, thanks for the catch.

**Review by akulakum — APPROVED** [review]

**Review by sokn-sys — APPROVED** [review]

**Pull request merged**